src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -157,9 +161,9 @@ public virtual void CleanUp()
             }
 
             // Get the list of available versions, ignoring null versions
-            _allPackageVersions = versions
+            _allPackageVersions = (await Task.WhenAll(versions
                 .Where(v => v?.Version != null)
-                .Select(v => v.Version)
+                .Select(async v => (v.Version, (await (v.PackageSearchMetadata?.GetDeprecationMetadataAsync() ?? Task.FromResult((PackageDeprecationMetadata)null))) != null))))
```

Comment:
can this be simplified? perhaps by creating a method?  `Task.FromResult` is usually in a method that returns a `Task` but is not async, but this lamba method is async. And the `!= null` at the end  is confusing. Plus the `ToList` on the line below isn't running on the same sequence that `Where` and `Select` are, so it would be good if they were indented differently to make it more clear. The code before was easy to understand, this new code is not.

Reply:
created a method as requested.